pkg/logger/consolepretty/pretty.go

```diff
@@ -163,13 +163,35 @@ type Config struct {
 	// With set to true:
 	// 	[INFO | 2006-01-02T15:04:05Z | example.go] Sample message.
 	DisableCallerLine bool
+
+	// CallerMaxLength will trim the caller file and line down to this length
+	// if set to a value of 1 or higher.
+	//
+	// When set to 0:
+	// 	[INFO | 2006-01-02T15:04:05Z | example.go:20] Sample message.
+	// With set to 10:
+	// 	[INFO | 2006-01-02T15:04:05Z | …ple.go:20] Sample message.
+	CallerMaxLength int
+
+	// CallerMinLength will pad the caller file and line with spaces so that it
+	// reaches the target character width.
+	//
+	// When set to 0:
+	// 	[INFO | 2006-01-02T15:04:05Z | example.go:20] Sample message.
+	// 	[INFO | 2006-01-02T15:04:05Z | test.go:20] Sample message.
+	// With set to 13:
+	// 	[INFO | 2006-01-02T15:04:05Z | example.go:20] Sample message.
+	// 	[INFO | 2006-01-02T15:04:05Z | test.go:20   ] Sample message.
```

Comment:
(Line 136 to 184)
All the examples above this line seem to use an outdated format.
This one should be
```suggestion
	// 	2006-01-02T15:04:05Z [INFO |test.go:20   ] Sample message.
```
I think.

It would be nice if there was a test and example with DisableDate set to false. I understand this might be a bit tricky considering the way printing the date is currently handled though.

Reply:
We can have example tests without any `// Output:` comment. That way it would only be testing that it compiles and not the functionality of it, but for the sake of documentation that might be nice?

Reply:
I think it would be nice for the sake of documentation, yeah.
Being able to assert the output would be great, maybe an internal function that lets you spoof the time value, and then the public function passes in time.Now to it when called? Not sure what the best way to go about it would be.

Probably best to leave it for another PR though. This one is ready imo!